truss/tests/test_model_inference.py

```diff
@@ -1854,28 +1831,60 @@ async def predict(self, inputs: Dict):
 
 @pytest.mark.asyncio
 @pytest.mark.integration
-async def test_websocket_endpoint():
+async def test_raise_predict_and_websocket_endpoint():
     model = """
-    import fastapi
-    from typing import Dict, AsyncGenerator
-
     class Model:
-        def __init__(self):
+        async def websocket(self, websocket):
             pass
 
-        def load(self):
+        async def predict(self, inputs):
             pass
+    """
+    with ensure_kill_all(), _temp_truss(model, "") as tr:
+        container = tr.docker_run(
+            local_port=8090, detach=True, wait_for_server_ready=False
+        )
+        time.sleep(1)
+        _assert_logs_contain_error(
+            container.logs(),
+            message="Exception while loading model",
+            error="cannot have both `predict` and `websocket` method",
+        )
+
 
+@pytest.mark.asyncio
+@pytest.mark.integration
+async def test_raise_no_endpoint():
+    model = """
+    class Model:
+       pass
+    """
+    with ensure_kill_all(), _temp_truss(model, "") as tr:
+        container = tr.docker_run(
+            local_port=8090, detach=True, wait_for_server_ready=False
+        )
+        time.sleep(1)
+        _assert_logs_contain_error(
+            container.logs(),
+            message="Exception while loading model",
+            error="must have a `predict` or `websocket` method",
+        )
+
+
+@pytest.mark.asyncio
+@pytest.mark.integration
+async def test_websocket_endpoint():
+    model = """
+    import fastapi
+
+    class Model:
         async def websocket(self, websocket: fastapi.WebSocket):
             try:
                 while True:
                     text = await websocket.receive_text()
                     await websocket.send_text(text + " pong")
-            except WebSocketDisconnect:
+            except fastapi.WebSocketDisconnect:
```

Comment:
If we're trying to make the underlying web socket opaque, should we also raise our own disconnect error?

Reply:
Since we don't have traditional trusses depend on the `truss` lib (yet) I think we have to pass that through. I'm torn on masking it for chains though:
- Benefit is that we don't leak the underlying impl (as you noted). However, there are probably lots of other errors that `fastapi` emits beyond the `WebSocketDisconnect`, and then we have to decide which ones to mask
- Drawback is that user has to read our docs to know this is a difference, rather than using their existing knowledge on how to use websockets (if coming from a fastapi world)

Reply:
That makes sense. Personally I'm not too keen on making it opaque - FastAPI seems like a safe long-term bet. But if we are making it opaque it'd be nice to be consistent.

Reply:
That sounds good to me! The wrapper as it stands today is more compelling to me for limiting potential websocket behavior, so being able to change the underlying impl later seems less important? If we really did need to swap out the FastAPI layer, I imagine we'd deprecate it via TaT rather than try to change it under the hood